entc/integration/integration_test.go

```diff
@@ -1406,11 +1406,46 @@ func Tx(t *testing.T, client *ent.Client) {
 		var m mocker
 		m.On("onRollback", nil).Once()
 		defer m.AssertExpectations(t)
-		tx.OnRollback(m.rHook())
+
+		tx.OnRollback(func(next ent.Rollbacker) ent.Rollbacker {
+			return ent.RollbackFunc(func(ctx context.Context, tx *ent.Tx) error {
+				err := next.Rollback(ctx, tx)
+				m.onRollback(err)
+
+				require.NotNil(t, ctx)
+
+				return err
+			})
+		})
 		err = tx.Item.Create().Exec(ctx)
-		require.Error(t, err)
+		require.Error(t, err) // because of read-only in tx opts
```

Comment:
Please, change it to an assertion message.
```suggestion
		require.Error(t, err, "expect creation to fail in read-only tx")
```